cpg-analysis/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/UnreachableEOGPass.kt

```diff
@@ -91,24 +96,34 @@ class UnreachableEOGPass(ctx: TranslationContext) : TranslationUnitPass(ctx) {
  * Returns the updated state and true because we always expect an update of the state.
  */
 fun transfer(
-    currentEdge: Edge<Node>,
-    currentState: State<Edge<Node>, Reachability>,
-): State<Edge<Node>, Reachability> {
+    lattice: Lattice<UnreachabilityStateElement>,
+    currentEdge: EvaluationOrder,
+    currentState: UnreachabilityStateElement,
+): UnreachabilityStateElement {
+    val lattice = lattice as? UnreachabilityState ?: return currentState
+    var newState = currentState
     when (val currentNode = currentEdge.end) {
         is IfStatement -> {
-            handleIfStatement(currentEdge, currentNode, currentState)
+            newState = handleIfStatement(lattice, currentEdge, currentNode, newState)
```

Comment:
While looking through the code I noticed that we have this explicit handlers for `IfStatement` and `WhileStatement`, but not for the other branching Constructs. is that intentional? Are they not Needed for for, foreach, do and switch?

Reply:
I'll double-check if some of these statements can be added. I think that some of these do not exist in the current implementation because the `ValueEvaluator` is still limited (this affects the foreach) and because I wasn't sure how to implement an appropriate comparison (this is the case for the switch).

Reply:
It should generalize to other loops as well. For the switch, we cannot easily identify a matching edge/case statement which is why I only added a TODO for this one.

Reply:
Added #2058